lib/sla.js

```diff
@@ -45,80 +48,135 @@ function designateMixingBridge(client, bridgeName) {
 }
 
 /**
- * Adds inbound and originated channels to the bridge.
+ * Updates the state of the shared extension.
  * @param {Object} client - Object that contains information from the ARI
  *   connection.
- * @param {Object} inbound - Incoming channel to be added to the bridge.
+ * @param {String} name - name of the shared extension to update.
+ * @param {String} state - state to update the shared extension to.
+ */
+function updateState(client, name, state) {
+  var deviceState = Q.denodeify(client.deviceStates.update.bind(client));
+  deviceState({deviceName: 'Stasis:' + name, deviceState: state});
+}
+
+/**
+ * Accesses the current state of the shared extension.
+ * @param {Object} client - Object that contains information from the ARI
+ *   connection.  Houses the device states
+ * @param {String} name - name of the shared extension to access.
+ * @return {String} - the current device state of this extension
+ */
+function getState(client, name) {
+  var getDeviceState = Q.denodeify(client.deviceStates.get.bind(client
+        .deviceStates));
+  return getDeviceState({deviceName: 'Stasis:' + name}).then(function (ds) {
+    return ds.state;
+  });
+}
+
+/**
+ * Adds the caller channel and originated channel (if applicable) to the bridge.
+ * @param {Object} client - the client (used for hanging up at the end).
+ * @param {Object} channel - Incoming channel to be added to the bridge.
  * @param {Object} dialed - Created/dialed channel to be added as well
  * @param {Object} bridge - Bridge that channels are to be added to.
+ * @param {Object} extension - the object data related to this sharedExtension
  * @return {Q} Q promise object.
  */
-function addChannelsToBridge(client, inbound, dialed, bridge) {
+function addChannelsToBridge(client, channel, dialed, bridge, extension) {
+  // The only way that the channel length will be 1 is if the outside user is
+  // still inside of the shared extension waiting for others to answer.
+  if (bridge.channels.length === 1) {
```

Comment:
I may be incorrect, but I think the only way that the bridge.channels.length could only be equal to one is if the outside channel was left hanging on hold. When an initial inbound or outbound call is made, two channels are always involved (insider and outsider). From there, only one station at a time can be added. So if there is only one channel left hanging, it has to be an outside channel. No outside channel means no call going on.